README.md

```diff
@@ -481,6 +485,69 @@ The verified in-toto statement may be written to stdout with the
 
 Note that `--source-uri` supports GitHub repository URIs like `github.com/$OWNER/$REPO` when the build was enabled with a Cloud Build [GitHub trigger](https://cloud.google.com/build/docs/automating-builds/github/build-repos-from-github). Otherwise, the build provenance will contain the name of the Cloud Storage bucket used to host the source files, usually of the form `gs://[PROJECT_ID]_cloudbuild/source` (see [Running build](https://cloud.google.com/build/docs/running-builds/submit-build-via-cli-api#running_builds)). We recommend using GitHub triggers in order to preserve the source provenance and valiate that the source came from an expected, version-controlled repository. You _may_ match on the fully-qualified tar like `gs://[PROJECT_ID]_cloudbuild/source/1665165360.279777-955d1904741e4bbeb3461080299e929a.tgz`.
 
+## Verification Summary Attestations (VSA)
+
+We have support for [verifying](https://slsa.dev/spec/v1.1/verification_summary#how-to-verify) VSAs.
+Rather than passing in filepaths as arguments, we allow passing in mulitple `--subject-digest` cli options, to
+accomodate subjects that are not simple-files.
+
+
+The verify-vsa command
```

Comment:
Why do you think this is dangerous? If you get to the point where you have a VSA you need to verify, I assume you should understand what the VSA represents. So maybe adding a link to https://slsa.dev/spec/v1.1/verification_summary#purpose to make sure users understand one?

Reply:
@laurentsimon Instead of `verify-artifact --vsa-path`, I was thinking to continue with `verify-vsa path1 path2 ...`, or `verify-vsa --artifact-path path1 --artifact-path path2 ...`

Reply:
i don't think re-using `verify-vsa` is consistent with the existing API. We currently use `verify-artifact` and `verify-image`, so I think it's the right way to build on existing APIs. `verify-vsa` is a low-level API for the rare use case that the artifact is not available

Reply:
re: why it's dangerous API. TOCTOU is an example of a vuln https://github.com/slsa-framework/slsa-verifier?tab=readme-ov-file#containers-1. We've been very careful to design APIs that prevent users from shooting themselves ini the foot. Leaving it up to users to calculate a sha is both poor UX and prone to mistakes.

Reply:
OK, I see what you're saying. This seems more like a not ideal UX rather than dangerous, since we are asking for a digest, and the digest _should_ be an immutable reference - if it's not, then it's technically non-compliant with SLSA. If we were to add a comment, I might say that we'd like to refactor this at some point to move VSA verification under `verify-artifact`.

Reply:
I think of the CLI options more like 

`verify-artifact` actually means "verify-provenance-with-artifact", or "verify-artifact-provenance"

With this thinking, a VSA is not actually a provenance (build provenance). And I'm open to making a new command "verify-artifact-vsa". To avoid the danger of the user not realizing that they're not enjoying the benefits of an actual build provenance.

re: TOCU, you're referring to how `verify-image` relies on the user using external tools to calculate digests (docs incorrectly says "tarball")
 - https://github.com/slsa-framework/slsa-verifier?tab=readme-ov-file#the-verify-image-command

Reply:
> verify-artifact actually means "verify-provenance-with-artifact", or "verify-artifact-provenance"

That was not the original intention, and that's why the term 'provenance' is not in the command's name. It's intended to verify artifacts with whatever attestation. It so happens the only one currently supported is provenance, but it need not be. If you ever need to verify an artifact with several attestations, you'll be able to do `verify-artifact --provenance-path --source-path etc`... unless they are packed into a bundle and you'll likely use `--attestations-path` like in npm command.

Reply:
> OK, I see what you're saying. This seems more like a not ideal UX rather than dangerous, since we are asking for a digest, and the digest _should_ be an immutable reference - if it's not, then it's technically non-compliant with SLSA. If we were to add a comment, I might say that we'd like to refactor this at some point to move VSA verification under `verify-artifact`.

just have to be careful that users don't use the non-immutable image for pulling